### PR TITLE
Fix serviceaccount continual regeneration by service account controller

### LINKED ISSUE #2573: How to create helm application in multi clusters
**Please provide an in-depth description of the question you have**:
I want to create applications in multiple clusters with argocd and karmada .I try to deploy a grafana by chart，but it keep creating sercret for serviceaccount 

```
grafana-token-xx9sq   kubernetes.io/service-account-token   3      3s
grafana-token-z29d5   kubernetes.io/service-account-token   3      7s
grafana-token-zlq2m   kubernetes.io/service-account-token   3      7s
grafana-token-zsrp7   kubernetes.io/service-account-token   3      9s
grafana-token-zxm25   kubernetes.io/service-account-token   3      14s
grafana-token-znxwf   kubernetes.io/service-account-token   3      0s
grafana-token-zpvnh   kubernetes.io/service-account-token   3      0s
grafana-token-zsx79   kubernetes.io/service-account-token   3      0s
grafana-token-s9mrp   kubernetes.io/service-account-token   3      0s
grafana-token-xhdnd   kubernetes.io/service-account-token   3      0s
grafana-token-9dst5   kubernetes.io/service-account-token   3      0s
grafana-token-6g2nc   kubernetes.io/service-account-token   3      0s
grafana-token-vwgh6   kubernetes.io/service-account-token   3      0s
grafana-token-f9h5w   kubernetes.io/service-account-token   3      0s
grafana-token-5m2dv   kubernetes.io/service-account-token   3      0s
grafana-token-xtbp9   kubernetes.io/service-account-token   3      0s
grafana-token-6qbk8   kubernetes.io/service-account-token   3      0s
grafana-token-kr7gl   kubernetes.io/service-account-token   3      0s
grafana-token-r7vp6   kubernetes.io/service-account-token   3      0s
grafana-token-9tmj2   kubernetes.io/service-account-token   3      0s
grafana-token-q2j9m   kubernetes.io/service-account-token   3      0s

[root@node1 test]# kubectl get secrets | wc -l
4981
[root@node1 test]# kubectl get secrets | wc -l
4991
[root@node1 test]# kubectl get secrets | wc -l
5002
[root@node1 test]# kubectl get secrets | wc -l
5013
```

1. I used the command to create it 

```
argocd app create grafana --repo https://charts.bitnami.com/bitnami --helm-chart grafana --revision 8.1.1 --dest-namespace default --dest-name karmada-apiserver --helm-set service.type=NodePort
```
微信图片

2. apply the propagation 

```
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: grafana
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      name: grafana
    - apiVersion: v1
      kind: Service
      name: grafana
    - apiVersion: v1
      kind: PersistentVolumeClaim
      name: grafana
    - apiVersion: v1
      kind: ConfigMap
      name: grafana-envvars
    - apiVersion: v1
      kind: Secret
      name: grafana-admin
    - apiVersion: v1
      kind: ServiceAccount
      name: grafana
  placement:
    clusterAffinity:
      clusterNames:
        - member1
```

**And I tried another way to verify the cause of the problem**

1. helm install bitnami/grafana --kubeconfig /etc/karmada/karmada-apiserver.config
2. apply the propagation

But the results are the same,I think it's Karmada, not argocd caused it 

**What do you think about this question?**:

How should I make the application correctly distributed. What's the problem with my method?

**Environment**:

- Karmada version: 1.2.0
- Kubernetes version: 1.22.9
- Others: grafana 8.1.1

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
For serviceAccount, we don't retain that in control plane but merge that in control plane and member clusters. You can based on https://github.com/karmada-io/karmada/issues/2573#issuecomment-1256844989.

**Which issue(s) this PR fixes**:
Fixes #2573 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

File: pkg/util/lifted/retain.go [hunk 1]
```diff
@@ -84,33 +84,51 @@ func retainServiceClusterIP(desired, observed *unstructured.Unstructured) error
...
-// ensures that the sync controller doesn't continually clear a generated
+// RetainServiceAccountFields merges the 'secrets' field between the control plane
+// and the member clusters and retains that of a service account. This
+// ensures that the karmada-controller-manager doesn't continually clear a generated
 // secret from a service account, prompting continual regeneration by the
 // service account controller in the member cluster.
 func RetainServiceAccountFields(desired, observed *unstructured.Unstructured) (*unstructured.Unstructured, error) {
-	// Check whether the secrets field is populated in the desired object.
+	var updatedSecrets []interface{}
+	isSecretExistMap := make(map[string]bool)
```
Comment (RainbowMango) [inline pkg/util/lifted/retain.go:94]: // map[string]struct{}
or
// sets.String()

File: pkg/util/lifted/retain_test.go [hunk 2]
```diff
@@ -187,3 +188,106 @@ func TestRetainClusterIPInServiceFields(t *testing.T) {
...
+		t.Run(test.name, func(t *testing.T) {
+			desiredObj, err := RetainServiceAccountFields(test.desiredObj, test.observedObj)
+			if (err != nil) != test.expectedErr {
+				t.Fatalf("test %s fails: unexpected returned error %v", test.name, err)
+			}
+
+			if err == nil {
+				currentSecretValue, ok, err := unstructured.NestedSlice(desiredObj.Object, SecretsField)
+				if err != nil {
+					t.Fatalf("test %s fails: %v", test.name, err)
```
Comment (RainbowMango) [inline pkg/util/lifted/retain_test.go:282]: Don't need to log `test name`.

Comment (Poor12) [reply pkg/util/lifted/retain_test.go]: done

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**